dao/src/main/java/org/wso2/testgrid/dao/repository/TestPlanRepository.java

```diff
@@ -195,16 +195,17 @@ public TestPlan getLastBuild(Product product) {
      * @return a list of {@link TestPlan}
      */
     public List<TestPlan> getLatestTestPlans(Product product) {
-        String sql = "select t.* from test_plan t inner join (select tp.infra_parameters, max(tp.modified_timestamp) AS"
-                + " time, dp.name , dp.id As dep_id from test_plan tp inner join deployment_pattern dp on "
-                + "tp.DEPLOYMENTPATTERN_id=dp.id  and tp.DEPLOYMENTPATTERN_id in "
-                + "(select id from deployment_pattern where PRODUCT_id= ? ) group by tp.infra_parameters,dp.id) as x "
-                + "on t.infra_parameters=x.infra_parameters AND t.modified_timestamp=x.time AND "
-                + "t.DEPLOYMENTPATTERN_id=x.dep_id;";
+        String sql = "select tp.* from test_plan tp inner join (Select distinct infra_parameters, max(test_run_number) "
+                + "as test_run_number from test_plan where  DEPLOYMENTPATTERN_id in (select id from deployment_pattern "
+                + "where PRODUCT_id= ?) group by infra_parameters) as latest_test_run_nums on "
+                + "tp.infra_parameters=latest_test_run_nums.infra_parameters and "
+                + "tp.test_run_number=latest_test_run_nums.test_run_number and tp.DEPLOYMENTPATTERN_id in "
+                + "(select id from deployment_pattern where PRODUCT_id= ?);";
```

Comment:
You definitely fixed something in here. I'm not clear what it is though. :-) Let's see whether we can simplify this.

Reply:
As per the discussion we had, this is fixed with 9be59e94046fe2feb8fe36c3e1d77913ba1669a7